### ISSUE TITLE
Are there backreferences / capture groups in Waxeye?

### ISSUE THREAD

**Issue by JoshuaGross** [orlandohill/waxeye#19]
Seems like an important feature for a PEG parser but I can't find anything about it?

**Comment by orlandohill** [comment]
Currently, Waxeye doesn't have backreferences or capture groups.

The grammar file that defines the Waxeye grammar language has a way to label the result of a parsing expression, and give those results to external code. The external code would then be used to do context sensitive checks beyond what a normal parsing expression grammar can support. However, I haven't implemented those features yet, because I have never had the need.

What problem are you trying to solve?

**Comment by JoshuaGross** [comment]
Basically HTML tag parsing: <tag></tag>
For now I'm just hoping that they match up; I could assert that they match, also. If they don't it's definitely user error. So not a critical issue. 

Sent from Outlook




On Wed, Aug 26, 2015 at 7:34 PM -0700, "Orlando Hill" <notifications@github.com> wrote:












Currently, Waxeye doesn't have backreferences or capture groups.



The grammar file that defines the Waxeye grammar language has a way to label the result of a parsing expression, and give those results to external code. The external code would then be used to do context sensitive checks beyond what a normal parsing expression grammar can support. However, I haven't implemented those features yet, because I have never had the need.



What problem are you trying to solve?

**Comment by orlandohill** [comment]
Yes, checking that the tag names match in another phase would be an easy solution. Specifically for HTML, I'd try to use preexisting libraries, if at all possible.

**Comment by JoshuaGross** [comment]
Yeah, it's not HTML so I do need a parser generator, but I hear ya.
